bluetooth: controller: Update slot reservation time calculation for ADV

erbr-ot opened the pull request:
Correct slot reservation time calc for legacy advertise
Add dependency on data lengths for advertise and scan response

erbr-ot committed:
bluetooth: controller: Add def's re. update to slot reservation calc

Add defines to support updated slot reservation calculation in advertise
Note: Numbers used in added defines are subject to HW specific tuning

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

mped-oticon commented on subsys/bluetooth/controller/ll_sw/ull_adv.c:
```diff
@@ -654,25 +650,68 @@ u8_t ll_adv_enable(u8_t enable)
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+	const u8_t phy = lll->phy_p;
+#else
+	const u8_t phy = 1;
```
"Normal advertise is always at 1Mbps"

erbr-ot replied:
Assuming this means suggestion to add a comment to this effect, I will do just that.

mped-oticon commented on subsys/bluetooth/controller/ll_sw/nordic/lll/lll_vendor.h:
```diff
@@ -5,8 +5,18 @@
+#define EVENT_RX_JITTER_US(phy) 16    /* Radio Rx timing uncertainty */
+#define EVENT_RX_TO_US(phy) (((((phy)&0x03) + 4)*8 / BIT((((phy)&0x3)>>1))) + \
+				  EVENT_RX_JITTER_US(phy))
+#define EVENT_RX_TX_TURNARROUND(phy)  ((phy) == 1?100:((phy) == 2 ? 80:900))
```
100,80,900 us: How sure are we (Oticon) about these (Nordic) timing constants?

erbr-ot replied:
Not at all. I was not sure how to handle that aspect, so I chose to simply add the numbers that Oticon uses, and expected for Nordic to change these to be correct.

cvinayak commented on subsys/bluetooth/controller/ll_sw/pdu.h:
```diff
@@ -437,14 +453,14 @@ struct pdu_data {
 	union {
 		struct pdu_data_llctrl llctrl;
-		u8_t                   lldata[0];
+		u8_t				lldata[0];
```
No use of tabs other than beginning of a line.

cvinayak reviewed (commented):
Need few days next week to complete review.

mped-oticon commented:
@erbr-ot In case you haven't already noticed, it seems like you are missing a file for defining `TARGETA_SIZE`.

erbr-ot commented:
@nashif CI is failing on boards that ought not be affected by the changes in this PR. Advice appreciated.

erbr-ot commented:
@carlescufi CI appear to be failing on boards that ought not be affected by the changes in this PR. I'd much appreciate advice on this.

mped-oticon commented on subsys/bluetooth/controller/ll_sw/pdu.h:
```diff
@@ -15,7 +15,19 @@
+#define PREAMBLE_SIZE(phy)      (phy&0x3)
+#define LL_HEADER_SIZE(phy)     (PREAMBLE_SIZE(phy) + PDU_AC_LL_HEADER_SIZE \
+				  + ACCESS_ADDR_SIZE + CRC_SIZE)
+#define BYTES2US(bytes, phy)    ((bytes)*8/BIT((phy&0x1)>>1))
```
nit: We should use `PKT_US` only.
`PKT_US` is now available for all to use (was limited to Nordic at first), which turns `BYTES2US` into duplicate code.

erbr-ot replied:
PKT_US() macro is local to ull_conn.c and could possibly be modified to utilize BYTES2US.

mped-oticon reviewed (changes requested):
LGTM, but have one nitpick.

cvinayak reviewed (changes requested):
Could you revert the whitespace changes, so that only actual change is under review.

erbr-ot committed:
bluetooth: controller: Update slot reservation time calculation for ADV

Correct slot reservation time calc for legacy advertise
Add dependency on data lengths for advertise and scan response

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

erbr-ot commented:
@cvinayak and @mped-oticon would you guys re-review, as I believe I have covered the issues mentioned.

erbr-ot committed:
bluetooth: controller: Update to use new EVENT_IFS_US define

Added a new define (EVENT_IFS_US) to pdu.h - this is now used instead of
previous TIFS_US from vendor specific header

Signed-off-by: Erik Brockhoff <erbr@oticon.com>

cvinayak commented on subsys/bluetooth/controller/ll_sw/nordic/lll/lll_vendor.h:
```diff
@@ -9,4 +9,16 @@
 #define EVENT_JITTER_US               16
+
+#define EVENT_RX_JITTER_US(phy) 16    /* Radio Rx timing uncertainty */
+#define EVENT_RX_TO_US(phy) (((((phy)&0x03) + 4)*8 / BIT((((phy)&0x3)>>1))) + \
```
replace *8  with <<3

cvinayak reviewed (changes requested):
one minor comment.

cvinayak commented:
@carlescufi this is ready for merge.

aescolar commented:
@carlescufi This PR is ready, could you please merge
